src/protocols/bdx/BdxTransferSession.cpp

```diff
@@ -848,13 +845,13 @@ void TransferSession::PrepareStatusReport(StatusCode code)
 {
     mStatusReportData.statusCode = code;
 
-    Encoding::LittleEndian::PacketBufferWriter bbuf(chip::MessagePacketBuffer::New(kStatusReportMinSize), kStatusReportMinSize);
-    VerifyOrReturn(!bbuf.IsNull());
+    Protocols::SecureChannel::StatusReport report(static_cast<uint16_t>(Protocols::SecureChannel::GeneralStatusCode::kFailure),
```

Comment:
Can we have StatusReport accept `Protocols::SecureChannel::GeneralStatusCode` for its first ctor arg?

Reply:
Sure, I'll change it. Definitely nicer to not have to force the caller to use `static_cast`